drakrun/drakrun/draksetup.py

```diff
@@ -245,6 +246,86 @@ def send_usage_report(report):
         logging.exception("Failed to send usage report. This is not a serious problem.")
 
 
+def create_rekall_profiles(install_info):
+    profiles_path = os.path.join(LIB_DIR, "profiles")
+
+    with tempfile.TemporaryDirectory() as mount_path:
+        # we mount 2nd partition, as 1st partition is windows boot related and 2nd partition is C:\\
+
+        if install_info["storage_backend"] == "zfs":
+            # workaround for not being able to mount a snapshot
+            base_snap = shlex.quote(os.path.join(install_info["zfs_tank_name"], 'vm-0@booted'))
+            tmp_snap = shlex.quote(os.path.join(install_info["zfs_tank_name"], 'tmp'))
+            try:
+                subprocess.check_output(f'zfs clone {base_snap} {tmp_snap}', shell=True)
+            except subprocess.CalledProcessError:
+                logging.warning("Failed to clone temporary zfs snapshot. Aborting generation of usermode rekall profiles")
+                return
+
+            tmp_mount = shlex.quote(os.path.join("/", "dev", "zvol", install_info["zfs_tank_name"], "tmp-part2"))
+            try:
+                # We have to wait for a moment for zvol to appear
+                time.sleep(1.0)
```

Comment:
This should be checked in a more reliable way. There is absolutely no guarantee that zvol would appear after 1 second and this problem was already experienced in production. I strongly suggest to check whether `tmp_mount` file exists in a loop (with some reasonable hard timeout like 60 seconds).

Reply:
Yup, I guess that looping should be more stable, however that's still not perfect. Maybe there's some way to ask ZFS to wait until the device appears?

Reply:
Fixed in https://github.com/CERT-Polska/drakvuf-sandbox/pull/148/commits/f7d849699637f7d4bc339403c2de769e77e141e8